### PR TITLE
Add delete flag to metadata repsonse

### PR DESCRIPTION
**Pull request by ClaytonPassmore**
@mattyu007 is this okay?

### CODE REVIEW

File: Spec/API.md [hunk 1]
```diff
@@ -355,7 +355,8 @@ This response only contains metadata (does not include the cards in the deck).
     "created": "<created>"
```
Comment (mattyu007) [inline Spec/API.md:355]: GitHub won't let me select this, but should `user_data-version` be `user_data_version`?

File: Spec/API.md [hunk 2]
```diff
@@ -373,6 +374,7 @@ Response description:
 - `last_update`: ISO format UTC datetime string describing the time of the most recent deck update
 - `last_update_device`: the name of the device that last modified the deck (this is only available for decks that are owned by the requester)
 - `share_code`: the code that can be entered by users to gain access to a remote private deck
+- `deleted`: boolean indicating whether or not a deck has been deleted by the owner
```
Comment (mattyu007) [inline Spec/API.md:377]: If `deleted` is true, should the client still expect to receive all the other metadata?

File: Spec/API.md (hunk 1 above)
Comment (mattyu007) [reply Spec/API.md:355]: Ditto with `PUT /api/v1/deck/<uuid>/flag`, where
```
 {
     user-data-version: <user-data-version>
 }
 ```
should be
```
 {
     "user_data_version": <user-data-version>
 }
 ```

Feel free to address these in a separate PR.

**Review by mattyu007 — APPROVED** [review]

File: Spec/API.md (hunk 2 above)
Comment (ClaytonPassmore) [reply Spec/API.md:377]: Probably not. I'll make a note

File: Spec/API.md (hunk 1 above)
Comment (ClaytonPassmore) [reply Spec/API.md:355]: Thanks for catching that, I'll throw those in here too

**Pull request merged**